src/components/SingleBlog/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from "react";
+import "./index.scss";
+import { Container, Col, Row } from "react-bootstrap";
+import { useParams } from "react-router-dom";
+import { ReactComponent as FacebookIcon } from "../../images/icons/facebookIcon.svg";
+import { ReactComponent as TwitterIcon } from "../../images/icons/twitterIcon.svg";
+import { ReactComponent as LinkedInIcon } from "../../images/icons/linkedInIcon.svg";
+import { ReactComponent as InstagramIcon } from "../../images/icons/instagramIcon.svg";
+
+const RELATED_POSTS_TO_SHOW = 4;
+
+export default function SingleBlog(props) {
+  const params = useParams();
+  const [blogPost, setBlogPosts] = useState();
+  const allBlogPosts = async () => {
+    const data = await fetch(
+      `https://public-api.wordpress.com/wp/v2/sites/safeplace102505649.wordpress.com/posts/${params.id}`
+    );
+    const posts = await data.json();
+    setBlogPosts(posts);
+  };
+  useEffect(() => {
+    allBlogPosts();
+  });
+
+  if (!blogPost) {
```

Comment:
I think this should be something else maybe but I dont know what :/

Reply:
it looks like it's waiting for the fetch to happen so it's undefined.

Reply:
this one here saying: if the blogPost is not loaded yet render loading when it's loaded show the blogpost data